data-extraction/pyDataExtraction/commonTypes/Text.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional
+from pyDataExtraction.commonTypes.base import DataType
+
+"""
+interface Text {
+	kind: { text: true };
+	text: string;
+	mimeType?: string;
+	fileName?: string;
+}
+"""
+
+
+class Text(DataType):
+    def __init__(
+        self,
+        text_data: str,
+        mimeType: Optional[str] = None,
+        fileName: Optional[str] = None,
+    ):
+        super().__init__()
+
+        self.kind["text"] = True
+        self.text = text_data
+        if mimeType is None:
+            self.mimeType = mimeType
+        if fileName is None:
```

Comment:
I suggest you wanted the opposite:
```
if file_name:
     self.file_name = file_name
```

Reply:
it may be when dealing with edge cases far removed from this or a leftover from python 2, but from what I keep reading, using `is None` seems to be preferred when dealing with values that may be `None`

Reply:
also, while editing this, I just remembered why I checked. because this library currently relies on `json.dumps(self.__dict__)` to print the json representation of an object in a format the plugin expects, I was only conditionally instantiating these variables.

they weren't part of the object unless they were explicitly added to the object. 

this may be another reason to either find a better method of creating json objects, or use super to get the inherited `json` representation and return a version altered in some way.